### PR TITLE
Add Patroni bin_name setting

### PR DESCRIPTION
**Pull request by pgstef**
## What Changed?

In Patroni 3.0.3 was added the `postgresql.bin_name.postgres` setting to simplify the EPAS support.

### CODE REVIEW

File: advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx [hunk 1]
```diff
@@ -167,7 +171,7 @@ EOF
 
 `$MY_IP` and `$MY_NAME` are specific to the local host. Otherwise, the `patroni.yml` configuration is the same on all Patroni nodes.
 
-Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Create a symbolic link to let Patroni find that binary:
+Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Before Patroni [version `3.0.3`](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), remove the `bin_name` setting from the configuration and create a symbolic link to let Patroni find that binary:
```
Comment (barthisrael) [inline advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]:
```suggestion
Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. As `postgresql.bin_name` setting was introduced in [version `3.0.3`](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), if you are running Patroni older than that, then remove the `bin_name` setting from the configuration and create a symbolic link to let Patroni find that binary:
```

I wonder if this extra verbosity makes it a bit more friendly?

File: advocacy_docs/supported-open-source/patroni/rhel8_quick_start.mdx [hunk 2]
```diff
@@ -169,7 +173,7 @@ EOF
 
 `$MY_IP` and `$MY_NAME` are specific to the local host. Otherwise, the `patroni.yml` configuration is the same on all Patroni nodes.
 
-Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Create a symbolic link to let Patroni find that binary:
+Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Before Patroni [version `3.0.3`](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), remove the `bin_name` setting from the configuration and create a symbolic link to let Patroni find that binary:
```
Comment (barthisrael) [inline advocacy_docs/supported-open-source/patroni/rhel8_quick_start.mdx:176]:
```suggestion
Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. As `postgresql.bin_name` setting was introduced in [version `3.0.3`](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), if you are running Patroni older than that, then remove the `bin_name` setting from the configuration and create a symbolic link to let Patroni find that binary:
```

Similar to previous one.

**Review by barthisrael — COMMENTED** [review]
Changes look good. I only proposed a small rephrase in the sentence describing `bin_name`.

**Review by theadamwright — DISMISSED** [review]

File: advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx (hunk 1 above)
Comment (pgstef) [reply advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]: Hm :thinking:, I don't really like to repeat the setting name again. Maybe:

_As the `postgresql.bin_name` setting was introduced in [version `3.0.3`](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), remove it from the configuration if you are running an older Patroni version and create a symbolic link..._

**Review by barthisrael — APPROVED** [review]
Looks great!

File: advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx [hunk 3]
```diff
@@ -167,7 +171,7 @@ EOF
 
 `$MY_IP` and `$MY_NAME` are specific to the local host. Otherwise, the `patroni.yml` configuration is the same on all Patroni nodes.
 
-Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Create a symbolic link to let Patroni find that binary:
+Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. As the `postgresql.bin_name` setting was introduced in the [`3.0.3` release](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), remove it from the configuration if you are running an older version and create a symbolic link to let Patroni find that binary:
```
Comment (martinmarques) [inline advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]: Specify that it's an older version of Patroni:
"... if you are running an older version of Patroni and create..."

File: advocacy_docs/supported-open-source/patroni/rhel8_quick_start.mdx [hunk 4]
```diff
@@ -169,7 +173,7 @@ EOF
 
 `$MY_IP` and `$MY_NAME` are specific to the local host. Otherwise, the `patroni.yml` configuration is the same on all Patroni nodes.
 
-Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. Create a symbolic link to let Patroni find that binary:
+Patroni expects to find the `postgres` binary in the `bin_dir` location, while the EDB Postgres Advanced Server binary is called `edb-postgres`. As the `postgresql.bin_name` setting was introduced in the [`3.0.3` release](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303), remove it from the configuration if you are running an older version and create a symbolic link to let Patroni find that binary:
```
Comment (martinmarques) [inline advocacy_docs/supported-open-source/patroni/rhel8_quick_start.mdx:176]: Specify that it's an older version of Patroni:
"... if you are running an older version of Patroni and create..."

**Review by martinmarques — CHANGES_REQUESTED** [review]

File: advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx (hunk 3 above)
Comment (martinmarques) [reply advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]: The order might be what confuses me there.
_The `postgresql.bin_name` setting was introduced in the [`3.0.3` release], so if you are running an older version of Patroni remove it from the configuration, and create a symbolic link to let Patroni find the server's binary:_
Still two Patroni's but a bit more sparse

Comment (djw-m) [reply advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]: Suggestion.

The `postgresql.bin_name` setting does not exist in Patroni prior to the [release of version 3.0.3]((https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303) and will be silently ignored by older versions. For these versions, use an appropriately named symbolic link in the `bin_dir` directory that points to the relevant Postgresql binary.

Comment (pgstef) [reply advocacy_docs/supported-open-source/patroni/debian11_quick_start.mdx:174]: I went for:

The `postgresql.bin_name` setting does not exist in Patroni prior to the [`3.0.3` release](https://github.com/zalando/patroni/blob/master/docs/releases.rst#version-303) and will be silently ignored by older versions. For these versions, create an appropriately named symbolic link that points to the relevant binary

**Review by martinmarques — APPROVED** [review]

**Pull request merged**